src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/HelpersWindows.cs

```diff
@@ -0,0 +1,367 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+using X509IssuerSerial = System.Security.Cryptography.Xml.X509IssuerSerial;
+
+using Microsoft.Win32.SafeHandles;
+
+using Internal.NativeCrypto;
+
+using static Interop.Crypt32;
+
+namespace Internal.Cryptography.Pal.Windows
+{
+    internal static class HelpersWindows
+    {
+        public static CryptographicException ToCryptographicException(this ErrorCode errorCode)
+        {
+            return ((int)errorCode).ToCryptographicException();
+        }
+
+        public static string ToStringAnsi(this IntPtr psz)
+        {
+            return Marshal.PtrToStringAnsi(psz);
+        }
+
+        // Used for binary blobs without internal pointers.
```

Comment:
We do. Kind of. The Resize() extension method in DEBUG builds makes sure you crash because it always moves the block in memory and renders interior pointers useless.